ion/reader.go

```diff
@@ -109,47 +109,49 @@ type Reader interface {
 	// stream.
 	StepOut() error
 
-	// BoolValue returns the current value as a boolean (if that makes sense). It returns
-	// an error if the current value is not an Ion bool.
-	BoolValue() (bool, error)
+	// BoolValue returns the current value as a boolean (if that makes sense). It returns nil
+	// if the value is null. If the current value is not an Ion bool, it returns an error.
```

Comment:
```suggestion
	// if the value is an Ion null. If the current value is not an Ion bool, it returns an error.
```

I am not sure if we need to make the distinction here since Go has `nil` and not `null`.